dask/array/reblock.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import absolute_import, division, print_function
+from itertools import count, product, chain, accumulate
```

Comment:
accumulate is not available in itertools in Python 2.  Suggest grabbing this from `toolz` instead